### PR TITLE
Reorganizing signal and inline functions for Net Sub-System

### PR DESCRIPTION
**Pull request by oknet**
This is the 1st step to merge PR #1131 into ATS 8.0.0.

### CODE REVIEW

**Comment by shinrich** [comment]
What is the motivation for moving the read_rescheduled, etc from being methods on UnixNetVConnection class to being free functions?

**Comment by oknet** [comment]
@shinrich 

The readDisable, read/writeReschedule and etcs are wrapper functions to read/write_disable, read/write_reschedule and etcs. 

These functions put netvc into the queue of NetHandler and update timeout counter for `InactivityCop` that is a part of `NetHandler`. I believe these functions are private APIs of `NetHandler` and should be a part of `NetHandler`.

In the `P_UnixNet.h`, It declared and defined private methods & functions for NetHandler.

**Comment by oknet** [comment]
There is `EThread::schedule_imm()` to put an `Event` into the Queue of EThread.
And the `read_reschedule()` put a NetVC into the Queue of NetHandler.

To define `NetHandler::read_reschedule()` is another idea here.

**Comment by bryancall** [comment]
I would leave these as methods instead of creating function calls.  I can see `read_reschedule` and `write_reschedule` being methods of the NetHandler.

**Comment by oknet** [comment]
@bryancall make these functions as methods of NetHandler but `netActivity()`.

The `read_ready_list` and `write_ready_list` are internal queue of NetHandler.
To operate on these 2 queues should be a method of NetHandler.

File: iocore/net/P_UnixNetVConnection.h [hunk 1]
```diff
@@ -208,13 +208,6 @@ class UnixNetVConnection : public NetVConnection
 
   virtual void net_read_io(NetHandler *nh, EThread *lthread);
   virtual int64_t load_buffer_and_write(int64_t towrite, MIOBufferAccessor &buf, int64_t &total_written, int &needs);
-  void readDisable(NetHandler *nh);
-  void readSignalError(NetHandler *nh, int err);
-  int readSignalDone(int event, NetHandler *nh);
-  int readSignalAndUpdate(int event);
```
Comment (oknet) [inline iocore/net/P_UnixNetVConnection.h]: There is no writeXXXX() functions defined here.

File: iocore/net/P_UnixNet.h [hunk 2]
```diff
@@ -389,30 +400,171 @@ check_transient_accept_error(int res)
   }
 }
 
+static inline void
+net_activity(UnixNetVConnection *vc, EThread *thread)
+{
+  Debug("socket", "net_activity updating inactivity %" PRId64 ", NetVC=%p", vc->inactivity_timeout_in, vc);
+  (void)thread;
```
Comment (SolidWallOfCode) [inline iocore/net/P_UnixNet.h]: I think this is better done by omitting the name from the parameter list, e.g.
```
net_activity(UnixNetVConnection * vc, EThread *)
```

File: iocore/net/P_UnixNet.h [hunk 3]
```diff
@@ -389,30 +400,171 @@ check_transient_accept_error(int res)
...
+  } else {
+    write_ready_list.remove(vc);
+  }
+}
+
+//
+// Signal an event
+//
+TS_INLINE int
+NetHandler::read_signal_and_update(int event, UnixNetVConnection *vc)
```
Comment (SolidWallOfCode) [inline iocore/net/P_UnixNet.h]: Do we want to have more consistency by always having the `UnixNetVConnection*` argument first?

File: iocore/net/P_UnixNet.h (hunk 2 above)
Comment (SolidWallOfCode) [inline iocore/net/P_UnixNet.h]: I also think `net_activity` was better as a method of `NetVConnection` or `UnixNetVConnection` as it always requires a instance of that class.

Comment (oknet) [inline iocore/net/P_UnixNet.h]: Put back the `net_activity()` to `UnixNetVConnection.cc`.
I will merge our new Timeout System later and Reorganizing the `net_activity()`.

**Comment by bryancall** [comment]
There hasn't been an update on this PR for over 60 days, so I am closing it for now.  Feel free to reopen the PR when you have updated it.

**Pull request closed without merging**